src/test/scala/firrtl/backends/experimental/smt/end2end/EndToEndSMTSpec.scala

```diff
@@ -132,7 +131,10 @@ abstract class EndToEndSMTBaseSpec extends AnyFlatSpec with Matchers {
     }
     val fir = firrtl.Parser.parse(src)
     val name = fir.main
-    val testDir = BackendCompilationUtilities.createTestDirectory("EndToEndSMT." + name)
+    // @todo rewrite BackendCompilationUtilities
```

Comment:
how hard would it be to do this as part of this PR?
Would love to see essentially one big PR that atomically changes everything to using os-lib

Reply:
It will be a huge refactor which contains new API and deprecation warnings, And that's the reason why I choose change smt library firstly.
I personally prefer to a series of PRs for a simple review phase(since I'm not quite confident with the API changes(additions)), and a atomic PR which merges those all PRs for simple merge phase.

Reply:
I will work on this this week.